In [ ]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
#import du fichier CSV Mapping_Qualite.csv

Mapping_Qualite = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/Mapping_Qualite.csv',decimal=".")

Mapping_Qualite.head()

,Unnamed: 0,Zone,Country Code,Numeric,Code zone,Disponibilité alimentaire (Kcal/personne/jour),Quantité Aliments_disponible Poules(Milliers tonnes),Export_Utile(Milliers tonnes),Taux de croissance_population [%],Emploi dans la production_vegetale_animale (1000No),Distance_Economique Pays_France(Km),Indice Stabilite Politique,Croissance du PIB % annuel,"Qualité Reglementaire [-2,5..+2,5]"
0,0,Aruba,ABW,533.0,22.0,NaN,NaN,NaN,16.50,NaN,7711.830,1.43,4.26,0.95
1,1,Afghanistan,AFG,4.0,2.0,5.0,254.0,-103.185185,78.88,1185.84,5489.888,-2.48,2.27,-1.27
2,2,Angola,AGO,24.0,7.0,35.0,5873.0,277.000000,87.92,5998.46,6363.600,-0.34,4.42,-0.76
3,3,Anguilla,AIA,660.0,258.0,NaN,NaN,NaN,30.92,NaN,6737.844,1.11,NaN,NaN
4,4,Albania,ALB,8.0,3.0,85.0,438.0,38.000000,-7.88,421.64,1461.321,0.18,3.96,0.17


In [ ]:
#import du fichier CSV Terre_agricole_bio.csv

df_terre_agricole = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/Terre_agricole_bio.csv',decimal=".")

df_terre_agricole.head()

,Zone,[%]surfaces agricoles biologiques/total des terres agricoles
0,Albania,0.06
1,Algeria,0.01
2,Andorra,0.01
3,Argentina,3.43
4,Armenia,0.03


In [ ]:
df_terre_agricole.loc[145]

,145
Zone,Turkey
[%]surfaces agricoles biologiques/total des terres agricoles,0.82


In [ ]:
#Adaptation des noms des pays du dataframe df_terre_agricole avec le Mapping_pays
pd.options.mode.copy_on_write = True
df_terre_agricole.loc[26,'Zone']= "China; mainland"
df_terre_agricole.loc[34,'Zone']= "Czechia"
df_terre_agricole.loc[35,'Zone']= "Côte d'Ivoire"
df_terre_agricole.loc[51,'Zone']= "French Guiana"
df_terre_agricole.loc[57,'Zone']= "Guadeloupe"
df_terre_agricole.loc[77,'Zone']= "Lao People's Democratic Republic"
df_terre_agricole.loc[90,'Zone']= "Martinique"
df_terre_agricole.loc[94,'Zone']= "Republic of Moldova"
df_terre_agricole.loc[121,'Zone']= "Réunion"
df_terre_agricole.loc[139,'Zone']= "China; Taiwan Province of"
df_terre_agricole.loc[145,'Zone']= "Türkiye"
df_terre_agricole.loc[149,'Zone']= "United Kingdom of Great Britain and Northern Ireland"
df_terre_agricole.loc[149]

,149
Zone,United Kingdom of Great Britain and Northern I...
[%]surfaces agricoles biologiques/total des terres agricoles,2.84


In [ ]:
# jointure des 2 dataframes sur  colonne Zone

Mapping_Finale = Mapping_Qualite.merge(df_terre_agricole, on=["Zone"], how='outer')
display(Mapping_Finale)

,Unnamed: 0,Zone,Country Code,Numeric,Code zone,Disponibilité alimentaire (Kcal/personne/jour),Quantité Aliments_disponible Poules(Milliers tonnes),Export_Utile(Milliers tonnes),Taux de croissance_population [%],Emploi dans la production_vegetale_animale (1000No),Distance_Economique Pays_France(Km),Indice Stabilite Politique,Croissance du PIB % annuel,"Qualité Reglementaire [-2,5..+2,5]",[%]surfaces agricoles biologiques/total des terres agricoles
0,1,Afghanistan,AFG,4.0,2.0,5.0,254.0,-103.185185,78.88,1185.84,5489.8880,-2.48,2.27,-1.27,NaN
1,4,Albania,ALB,8.0,3.0,85.0,438.0,38.000000,-7.88,421.64,1461.3210,0.18,3.96,0.17,0.06
2,63,Algeria,DZA,12.0,4.0,22.0,4858.0,2.000000,36.04,NaN,1158.2920,-0.58,3.30,-0.95,0.01
3,10,American Samoa,ASM,16.0,5.0,NaN,NaN,NaN,-4.07,NaN,NaN,1.11,1.74,0.51,NaN
4,5,Andorra,AND,20.0,6.0,NaN,NaN,NaN,17.76,NaN,426.8953,1.58,3.37,1.35,0.01
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
240,238,Wallis and Futuna Islands,WLF,876.0,243.0,NaN,NaN,NaN,-20.64,NaN,16302.5200,NaN,NaN,NaN,NaN
241,67,Western Sahara,ESH,732.0,205.0,NaN,NaN,NaN,80.63,NaN,2663.3480,NaN,NaN,NaN,NaN
242,241,Yemen,YEM,887.0,249.0,30.0,282.0,78.000000,63.70,NaN,5235.1710,-2.56,0.75,-1.84,NaN
243,243,Zambia,ZMB,894.0,251.0,11.0,551.0,11.000000,66.59,3653.36,7221.1970,0.20,4.04,-0.50,0.01


In [ ]:
#Suppression de la colonne Unnamed: 0
Mapping_Finale = Mapping_Finale.drop('Unnamed: 0', axis=1)

In [ ]:
#vérification des valeurs manquantes
print(Mapping_Finale.isnull().sum())

Zone                                                              0
Country Code                                                      0
Numeric                                                           0
Code zone                                                         4
Disponibilité alimentaire (Kcal/personne/jour)                   73
Quantité Aliments_disponible Poules(Milliers tonnes)             71
Export_Utile(Milliers tonnes)                                    73
Taux de croissance_population [%]                                 9
Emploi dans la production_vegetale_animale (1000No)             103
Distance_Economique Pays_France(Km)                              19
Indice Stabilite Politique                                       30
Croissance du PIB % annuel                                       29
Qualité Reglementaire [-2,5..+2,5]                               38
[%]surfaces agricoles biologiques/total des terres agricoles     85
dtype: int64


In [ ]:
#Détail des valeurs manquantes pour la Code zone
Mapping_Finale.loc[Mapping_Finale['Code zone'].isnull(),:]

,Zone,Country Code,Numeric,Code zone,Disponibilité alimentaire (Kcal/personne/jour),Quantité Aliments_disponible Poules(Milliers tonnes),Export_Utile(Milliers tonnes),Taux de croissance_population [%],Emploi dans la production_vegetale_animale (1000No),Distance_Economique Pays_France(Km),Indice Stabilite Politique,Croissance du PIB % annuel,"Qualité Reglementaire [-2,5..+2,5]",[%]surfaces agricoles biologiques/total des terres agricoles
49,Christmas Island,CXR,162.0,NaN,NaN,NaN,NaN,NaN,NaN,11866.10,NaN,NaN,NaN,NaN
50,Cocos Islands,CCK,166.0,NaN,NaN,NaN,NaN,NaN,NaN,11388.36,NaN,NaN,NaN,NaN
162,Norfolk Island,NFK,574.0,NaN,NaN,NaN,NaN,NaN,NaN,17659.58,NaN,NaN,NaN,NaN
175,Pitcairn,PCN,612.0,NaN,NaN,NaN,NaN,NaN,NaN,15152.12,NaN,NaN,NaN,NaN


In [ ]:
#Renommage des colonnes--> plus propre
Mapping_Finale = Mapping_Finale.rename(columns={'Disponibilité alimentaire (Kcal/personne/jour)':'Disponibilité alimentaire en Viande de Volaille (Kcal/personne/jour)'})

In [ ]:
#Afficher les lignes où toutes les valeurs sont NaN

colonnes = ["Disponibilité alimentaire en Viande de Volaille (Kcal/personne/jour)", "Quantité Aliments_disponible Poules(Milliers tonnes)","Export_Utile(Milliers tonnes)" ,"Taux de croissance_population [%]",
           "Emploi dans la production_vegetale_animale (1000No)","Distance_Economique Pays_France(Km)","Indice Stabilite Politique",
            "Croissance du PIB % annuel","Qualité Reglementaire [-2,5..+2,5]","[%]surfaces agricoles biologiques/total des terres agricoles"]

Mapping_Finale[Mapping_Finale[colonnes].isna().all(axis=1)]



,Zone,Country Code,Numeric,Code zone,Disponibilité alimentaire en Viande de Volaille (Kcal/personne/jour),Quantité Aliments_disponible Poules(Milliers tonnes),Export_Utile(Milliers tonnes),Taux de croissance_population [%],Emploi dans la production_vegetale_animale (1000No),Distance_Economique Pays_France(Km),Indice Stabilite Politique,Croissance du PIB % annuel,"Qualité Reglementaire [-2,5..+2,5]",[%]surfaces agricoles biologiques/total des terres agricoles
48,Chine,CN,159.0,351.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN


In [ ]:
#Suppression des colonnes Numeric et Code zone

Mapping_Finale.drop(columns=['Numeric', 'Code zone'], inplace=True)

In [ ]:
#Ajout d'une colonne pour calculer le nb de NaN dans chaque ligne
Mapping_Finale = Mapping_Finale.assign(nb_nan=Mapping_Finale[colonnes].isna().sum(axis=1))
Mapping_Finale.head()

,Zone,Country Code,Disponibilité alimentaire en Viande de Volaille (Kcal/personne/jour),Quantité Aliments_disponible Poules(Milliers tonnes),Export_Utile(Milliers tonnes),Taux de croissance_population [%],Emploi dans la production_vegetale_animale (1000No),Distance_Economique Pays_France(Km),Indice Stabilite Politique,Croissance du PIB % annuel,"Qualité Reglementaire [-2,5..+2,5]",[%]surfaces agricoles biologiques/total des terres agricoles,nb_nan
0,Afghanistan,AFG,5.0,254.0,-103.185185,78.88,1185.84,5489.8880,-2.48,2.27,-1.27,NaN,1
1,Albania,ALB,85.0,438.0,38.000000,-7.88,421.64,1461.3210,0.18,3.96,0.17,0.06,0
2,Algeria,DZA,22.0,4858.0,2.000000,36.04,NaN,1158.2920,-0.58,3.30,-0.95,0.01,1
3,American Samoa,ASM,NaN,NaN,NaN,-4.07,NaN,NaN,1.11,1.74,0.51,NaN,6
4,Andorra,AND,NaN,NaN,NaN,17.76,NaN,426.8953,1.58,3.37,1.35,0.01,4


In [ ]:
Mapping_Finale.shape

(245, 13)

In [ ]:
# La fréquence des Nan dans notre dataframe

frequences = Mapping_Finale.loc[:, 'nb_nan'].value_counts()

# Convertir la série en DataFrame pour un affichage plus lisible
frequences_df = frequences.reset_index()
frequences_df.columns = ['nb_nan', 'Frequence']

# Afficher le tableau des fréquences
print(frequences_df)

    nb_nan  Frequence
0        0        106
1        1         47
2        2         16
3        5         15
4        8         13
5        4         12
6        6         11
7        3          9
8        9          8
9        7          7
10      10          1


Lors de la fusion de tous les dataframes,on a eu 245 pays au totale.
on a voulu connaître la fréquence de l’apparition des valeurs manquantes,
on a eu  :
Un échantillon avec 106 pays sans aucune données manquantes ⟶ 43%
Un échantillon avec 153 pays avec
    47 NaN ⟶ 62%
⟶ Choix de l’échantillon avec 153 pays tout en nettoyant les valeurs manquantes


In [ ]:
Donnees_totales = Mapping_Finale
Donnees_totales.head()

,Zone,Country Code,Disponibilité alimentaire en Viande de Volaille (Kcal/personne/jour),Quantité Aliments_disponible Poules(Milliers tonnes),Export_Utile(Milliers tonnes),Taux de croissance_population [%],Emploi dans la production_vegetale_animale (1000No),Distance_Economique Pays_France(Km),Indice Stabilite Politique,Croissance du PIB % annuel,"Qualité Reglementaire [-2,5..+2,5]",[%]surfaces agricoles biologiques/total des terres agricoles,nb_nan
0,Afghanistan,AFG,5.0,254.0,-103.185185,78.88,1185.84,5489.8880,-2.48,2.27,-1.27,NaN,1
1,Albania,ALB,85.0,438.0,38.000000,-7.88,421.64,1461.3210,0.18,3.96,0.17,0.06,0
2,Algeria,DZA,22.0,4858.0,2.000000,36.04,NaN,1158.2920,-0.58,3.30,-0.95,0.01,1
3,American Samoa,ASM,NaN,NaN,NaN,-4.07,NaN,NaN,1.11,1.74,0.51,NaN,6
4,Andorra,AND,NaN,NaN,NaN,17.76,NaN,426.8953,1.58,3.37,1.35,0.01,4


In [ ]:
df_full_data =  Donnees_totales.loc[(Donnees_totales['nb_nan']== 0) | (Donnees_totales['nb_nan']== 1) ]
display(df_full_data)

,Zone,Country Code,Disponibilité alimentaire en Viande de Volaille (Kcal/personne/jour),Quantité Aliments_disponible Poules(Milliers tonnes),Export_Utile(Milliers tonnes),Taux de croissance_population [%],Emploi dans la production_vegetale_animale (1000No),Distance_Economique Pays_France(Km),Indice Stabilite Politique,Croissance du PIB % annuel,"Qualité Reglementaire [-2,5..+2,5]",[%]surfaces agricoles biologiques/total des terres agricoles,nb_nan
0,Afghanistan,AFG,5.0,254.0,-103.185185,78.88,1185.84,5489.888,-2.48,2.27,-1.27,NaN,1
1,Albania,ALB,85.0,438.0,38.000000,-7.88,421.64,1461.321,0.18,3.96,0.17,0.06,0
2,Algeria,DZA,22.0,4858.0,2.000000,36.04,NaN,1158.292,-0.58,3.30,-0.95,0.01,1
5,Angola,AGO,35.0,5873.0,277.000000,87.92,5998.46,6363.600,-0.34,4.42,-0.76,NaN,1
8,Argentina,ARG,182.0,16332.0,-199.000000,20.32,68.22,10921.860,-0.13,-1.72,-0.48,3.43,0
...,...,...,...,...,...,...,...,...,...,...,...,...,...
237,Vanuatu,VUT,40.0,1.0,-128.185185,58.23,30.84,16440.020,0.93,3.97,-0.12,3.44,0
238,Venezuela (Bolivarian Republic of),VEN,71.0,1020.0,25.000000,19.41,1079.05,7736.347,-1.19,3.69,-2.03,0.01,0
239,Viet Nam,VNM,48.0,8824.0,290.000000,19.57,11703.14,9869.467,-0.04,7.09,-0.38,1.41,0
243,Zambia,ZMB,11.0,551.0,11.000000,66.59,3653.36,7221.197,0.20,4.04,-0.50,0.01,0


In [ ]:
df_full_data = df_full_data.reset_index(drop=True)

In [ ]:
# supprimer la colonne nb_nan
df_full_data = df_full_data.drop('nb_nan', axis=1)

In [ ]:
# supprimer la colonne Country Code
df_full_data = df_full_data.drop('Country Code', axis=1)

In [ ]:
#Remplissage des NaN de la colonne [%]surfaces agricoles biologiques/total des terres agricoles par 0
#Les valeurs que nous connaîssons pas, nous la mettrons à 0

df_full_data["[%]surfaces agricoles biologiques/total des terres agricoles"] = df_full_data["[%]surfaces agricoles biologiques/total des terres agricoles"].fillna(0)

df_full_data.head()

,Zone,Disponibilité alimentaire en Viande de Volaille (Kcal/personne/jour),Quantité Aliments_disponible Poules(Milliers tonnes),Export_Utile(Milliers tonnes),Taux de croissance_population [%],Emploi dans la production_vegetale_animale (1000No),Distance_Economique Pays_France(Km),Indice Stabilite Politique,Croissance du PIB % annuel,"Qualité Reglementaire [-2,5..+2,5]",[%]surfaces agricoles biologiques/total des terres agricoles
0,Afghanistan,5.0,254.0,-103.185185,78.88,1185.84,5489.888,-2.48,2.27,-1.27,0.00
1,Albania,85.0,438.0,38.000000,-7.88,421.64,1461.321,0.18,3.96,0.17,0.06
2,Algeria,22.0,4858.0,2.000000,36.04,NaN,1158.292,-0.58,3.30,-0.95,0.01
3,Angola,35.0,5873.0,277.000000,87.92,5998.46,6363.600,-0.34,4.42,-0.76,0.00
4,Argentina,182.0,16332.0,-199.000000,20.32,68.22,10921.860,-0.13,-1.72,-0.48,3.43


In [ ]:
#Verification de type des variables
df_full_data.dtypes

,0
Zone,object
Disponibilité alimentaire en Viande de Volaille (Kcal/personne/jour),float64
Quantité Aliments_disponible Poules(Milliers tonnes),float64
Export_Utile(Milliers tonnes),float64
Taux de croissance_population [%],float64
Emploi dans la production_vegetale_animale (1000No),float64
Distance_Economique Pays_France(Km),float64
Indice Stabilite Politique,float64
Croissance du PIB % annuel,float64
"Qualité Reglementaire [-2,5..+2,5]",float64


In [ ]:
#vérification des valeurs manquantes
print(df_full_data.isnull().sum())

Zone                                                                     0
Disponibilité alimentaire en Viande de Volaille (Kcal/personne/jour)     0
Quantité Aliments_disponible Poules(Milliers tonnes)                     0
Export_Utile(Milliers tonnes)                                            0
Taux de croissance_population [%]                                        0
Emploi dans la production_vegetale_animale (1000No)                     32
Distance_Economique Pays_France(Km)                                      0
Indice Stabilite Politique                                               0
Croissance du PIB % annuel                                               0
Qualité Reglementaire [-2,5..+2,5]                                       0
[%]surfaces agricoles biologiques/total des terres agricoles             0
dtype: int64


In [ ]:
#Détail des valeurs manquantes pour la variable Emploi dans la production_vegetale_animale (1000No)
df_full_data.loc[df_full_data['Emploi dans la production_vegetale_animale (1000No)'].isnull(),:]

,Zone,Disponibilité alimentaire en Viande de Volaille (Kcal/personne/jour),Quantité Aliments_disponible Poules(Milliers tonnes),Export_Utile(Milliers tonnes),Taux de croissance_population [%],Emploi dans la production_vegetale_animale (1000No),Distance_Economique Pays_France(Km),Indice Stabilite Politique,Croissance du PIB % annuel,"Qualité Reglementaire [-2,5..+2,5]",[%]surfaces agricoles biologiques/total des terres agricoles
2,Algeria,22.0,4858.0,2.000000,36.04,NaN,1158.292,-0.58,3.30,-0.95,0.01
8,Azerbaijan,44.0,1518.0,27.000000,22.49,NaN,3707.515,-0.73,4.07,-0.11,0.02
14,Belize,110.0,63.0,0.000000,54.89,NaN,8714.441,0.59,8.15,-0.43,0.24
23,Cameroon,11.0,2219.0,0.000000,62.54,NaN,4587.788,-1.40,3.67,-0.91,0.03
24,Canada,154.0,22704.0,19.000000,21.20,NaN,6326.645,0.82,1.53,1.64,2.26
27,China; mainland,59.0,259324.0,-124.000000,10.62,NaN,8716.882,-0.51,4.98,-0.36,0.66
29,Congo,72.0,52.0,104.000000,67.69,NaN,5746.042,0.02,2.58,-1.24,0.05
32,Cuba,82.0,1377.0,179.814815,1.90,NaN,7711.314,0.38,-1.93,-1.50,0.08
37,Dominica,150.0,0.0,4.000000,2.84,NaN,6829.773,1.29,2.05,0.31,11.63
45,Fiji,97.0,8.0,3.000000,8.94,NaN,16780.160,0.74,3.83,-0.09,4.13


In [ ]:
#recherche sur des années antérieurs l'exsitance des valeurs Emploi dans la production_vegetale_animale (1000No)
#Remplissage des valeurs NaN de Emploi dans la production_vegetale_animale (1000No) par leur valeur si elle existe
#Au début, nous avons exploité des données datant de 2017 à 2024 après nous avons pris pour toutes les années existantes
#sinon, remplissage par 0

df_full_data.loc[2,'Emploi dans la production_vegetale_animale (1000No)']= 0
df_full_data.loc[8,'Emploi dans la production_vegetale_animale (1000No)']= 0
df_full_data.loc[14,'Emploi dans la production_vegetale_animale (1000No)']= 0
df_full_data.loc[23,'Emploi dans la production_vegetale_animale (1000No)']= 0
df_full_data.loc[24,'Emploi dans la production_vegetale_animale (1000No)']= 0
df_full_data.loc[27,'Emploi dans la production_vegetale_animale (1000No)']= 0
df_full_data.loc[29,'Emploi dans la production_vegetale_animale (1000No)']= 38.66
df_full_data.loc[32,'Emploi dans la production_vegetale_animale (1000No)']= 0
df_full_data.loc[37,'Emploi dans la production_vegetale_animale (1000No)']= 0
df_full_data.loc[45,'Emploi dans la production_vegetale_animale (1000No)']= 90.24
df_full_data.loc[57,'Emploi dans la production_vegetale_animale (1000No)']= 0
df_full_data.loc[70,'Emploi dans la production_vegetale_animale (1000No)']= 0
df_full_data.loc[73,'Emploi dans la production_vegetale_animale (1000No)']= 0
df_full_data.loc[83,'Emploi dans la production_vegetale_animale (1000No)']= 0
df_full_data.loc[84,'Emploi dans la production_vegetale_animale (1000No)']= 0
df_full_data.loc[92,'Emploi dans la production_vegetale_animale (1000No)']= 0
df_full_data.loc[93,'Emploi dans la production_vegetale_animale (1000No)']= 0
df_full_data.loc[98,'Emploi dans la production_vegetale_animale (1000No)']= 0
df_full_data.loc[99,'Emploi dans la production_vegetale_animale (1000No)']= 806.17
df_full_data.loc[104,'Emploi dans la production_vegetale_animale (1000No)']= 0
df_full_data.loc[107,'Emploi dans la production_vegetale_animale (1000No)']= 0
df_full_data.loc[112,'Emploi dans la production_vegetale_animale (1000No)']= 0
df_full_data.loc[113,'Emploi dans la production_vegetale_animale (1000No)']= 0
df_full_data.loc[115,'Emploi dans la production_vegetale_animale (1000No)']= 0
df_full_data.loc[117,'Emploi dans la production_vegetale_animale (1000No)']= 0
df_full_data.loc[119,'Emploi dans la production_vegetale_animale (1000No)']= 0
df_full_data.loc[120,'Emploi dans la production_vegetale_animale (1000No)']= 0
df_full_data.loc[126,'Emploi dans la production_vegetale_animale (1000No)']= 132.17
df_full_data.loc[131,'Emploi dans la production_vegetale_animale (1000No)']= 12.23
df_full_data.loc[134,'Emploi dans la production_vegetale_animale (1000No)']= 613.54
df_full_data.loc[139,'Emploi dans la production_vegetale_animale (1000No)']= 4534.18
df_full_data.loc[147,'Emploi dans la production_vegetale_animale (1000No)']= 0


In [ ]:
#vérification des valeurs manquantes
print(df_full_data.isnull().sum())

Zone                                                                    0
Disponibilité alimentaire en Viande de Volaille (Kcal/personne/jour)    0
Quantité Aliments_disponible Poules(Milliers tonnes)                    0
Export_Utile(Milliers tonnes)                                           0
Taux de croissance_population [%]                                       0
Emploi dans la production_vegetale_animale (1000No)                     0
Distance_Economique Pays_France(Km)                                     0
Indice Stabilite Politique                                              0
Croissance du PIB % annuel                                              0
Qualité Reglementaire [-2,5..+2,5]                                      0
[%]surfaces agricoles biologiques/total des terres agricoles            0
dtype: int64


In [ ]:
df_full_data.shape

(153, 11)

In [ ]:
#préparation de l'export du dataframe en fichier CSV

df_full_data.to_csv('/content/drive/MyDrive/Colab Notebooks/Donnees_plus.csv')